ziggurat-core-geoip/src/providers/ip2loc.rs

```diff
@@ -21,8 +21,12 @@ impl Ip2LocationService {
 #[async_trait]
 impl GeoIPService for Ip2LocationService {
     async fn lookup(&self, ip: IpAddr) -> Result<GeoIPInfo, String> {
-        let mut db = DB::from_file(self.database_file).expect("failed to open database");
-        let record = db.ip_lookup(ip);
+        let db = DB::from_file(self.database_file);
```

Comment:
error handling and unwrapping can be done with one line with `map_err` followed by a `?`

Reply:
Done.